assets/src/edit-story/output/utils/getPreloadResources.js

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * AMP Extension
+ *
+ * @typedef {Object} PreloadResource
+ * @property {string} url The URL to the resource.
+ * @property {string} type Type of resource, e.g. "video" or "image".
+ */
+
+/**
+ * Goes through all pages in a story to find the resources that should be preloaded.
+ *
+ * Currently this includes the first page's background image/video.
+ *
+ * @param {Array} pages List of pages.
+ * @return {Array<PreloadResource>} List of preload resources.
+ */
+function getPreloadResources(pages) {
+  const preloadResources = [];
+
+  if (pages.length === 0) {
+    return preloadResources;
+  }
+
+  for (const { type, resource, isBackground } of pages[0].elements) {
```

Comment:
Should this be just the first page or the first page that has rich media?

Reply:
Great question!

Just the first page overall. Media on any pages beyond that will be preloaded by the AMP runtime as the user navigates through the story.